aptos-move/mvhashmap/src/versioned_group_data.rs

```diff
@@ -224,15 +247,22 @@ impl<T: Hash + Clone + Debug + Eq + Serialize, V: TransactionWrite> VersionedGro
         }
     }
 
-    fn remove(&mut self, shifted_idx: ShiftedTxnIndex) -> Vec<T> {
+    // Replacing is true when called from 'write' and false when called from pub remove (case when
+    // a write from a prior speculative execution is being fully removed).
+    fn remove(&mut self, shifted_idx: ShiftedTxnIndex, replacing: bool) -> Vec<(T, Option<usize>)> {
         // Remove idx updates first, then entries.
-        let idx_update_tags: Vec<T> = self
+        let idx_update_tags: Vec<(T, Option<usize>)> = self
             .idx_to_update
             .remove(&shifted_idx)
-            .map_or(vec![], |map| map.into_inner().into_keys().collect());
+            .map_or(vec![], |map| {
+                map.into_inner()
+                    .into_iter()
+                    .map(|(tag, v)| (tag, v.bytes_len()))
```

Comment:
Why map if we skip the second entry anyway in `for (tag, _) in idx_update_tags.iter() { ...`?

Reply:
we return it, we need to check whether sizes changed.